feat: add a setFormatter static function to format log entries for development

nm2501 opened the pull request:
This can be useful for developers running the application locally and presenting logs in a more developer-friendly form than the default stringified JSON.

nm2501 committed:
Added a setFormatter static function to Logger. This can be useful for developers running the application locally and presenting logs in a more developer-friendly form than the default stringified JSON.

igrek8 commented on src/Logger.ts:
```diff
@@ -52,6 +62,15 @@ export class Logger implements ILogger {
+   */
+  public static setFormatter(formatter?: LogFormatter) {
+    this.formatter = formatter || toJSON
+  }
```
```suggestion
  public static setFormatter(formatter: LogFormatter = toJSON) {
    this.formatter = formatter;
  }
```

nm2501 replied:
Done

igrek8 reviewed (changes requested):
Hello and thank you for creating this PR. I left a small suggestion. I will appreciate if you can fix the test coverage and then we can merge :)

igrek8 commented on src/Logger.ts:
```diff
@@ -10,6 +10,11 @@ import { toJSON } from './utils/toJSON';
+/**
+ * A function type which can format a log entry to a string
+ */
+type LogFormatter = (entry: LogEntry) => string
```
```suggestion
export type LogFormatter = (entry: LogEntry) => string;
```

igrek8 replied:
Let's re-export this type in `index.ts` to make it available for use. You can also create a file for this type alone.

nm2501 replied:
Done.

nm2501 committed:
Addressed some review comments

nm2501 commented:
> Hello and thank you for creating this PR. I left a small suggestion. I will appreciate if you can fix the test coverage and then we can merge :)

No worries. Happy to contribute. Thanks for making this module available.
I have addressed your comments but not sure what the issue with the test coverage is. Looks like 100% coverage already.
